src/nccl_ofi_rdma.c

```diff
@@ -1225,14 +1232,16 @@ static inline int handle_bounce_recv(struct fi_cq_tagged_entry *cq_entry, int ra
 		nccl_net_ofi_rdma_send_comm_t *s_comm = (nccl_net_ofi_rdma_send_comm_t *)comm;
 		assert(s_comm->local_comm_id == local_comm_id);
 		assert(bounce_data->recv_len == sizeof(nccl_net_ofi_rdma_ctrl_msg_t));
-
-		return handle_ctrl_recv(s_comm, msg_seq_num, bounce_req, ep);
+                int ret = handle_ctrl_recv(s_comm, msg_seq_num, bounce_req, ep);
+                NCCL_OFI_TRACE_POP();
+                return ret;
 	} else if (comm->type == NCCL_NET_OFI_RECV_COMM) {
 		/* Eager message */
 		NCCL_OFI_TRACE_EAGER_RECV(comm->dev_id, rail_id, comm, msg_seq_num);
 		nccl_net_ofi_rdma_recv_comm_t *r_comm = (nccl_net_ofi_rdma_recv_comm_t *)comm;
-
-		return handle_eager_recv(r_comm, msg_seq_num, bounce_req, ep);
+                int ret = handle_eager_recv(r_comm, msg_seq_num, bounce_req, ep);
+                NCCL_OFI_TRACE_POP();
```

Comment:
The current LTTNG are really just single points, whereas it seems like NVTX wants to measure ranges. I think we need to think about what we really want to measure here. As a user looking at the traces in NVTX, it wouldn't be immediately clear to me that, for example, the "eager_recv" trace spans the duration of the call to `handle_eager_recv`.